infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -600,6 +610,26 @@ public void accept(ProbeResult probeResult) {
     }
   }
 
+  /** Listens container's events and terminates runtime if unrecoverable event occurs. */
```

Comment:
AFAIK we don't have runtime interruption in place for k8s runtime yet. Can you elaborate how this handler stops start of a runtime?

Reply:
s/runtime/workspace/